Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -239,11 +239,13 @@ void MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessi
 {
     if (handler) {
         ALWAYS_LOG(LOGIDENTIFIER, "adding ", action);
+        Locker lock { m_actionHandlersLock };
         m_actionHandlers.set(action, handler);
         auto platformCommand = platformCommandForMediaSessionAction(action);
         if (platformCommand != PlatformMediaSession::NoCommand)
             PlatformMediaSessionManager::sharedManager().addSupportedCommand(platformCommand);
     } else {
+        Locker lock { m_actionHandlersLock };
```

Comment:
ditto

Reply:
Should I be concerned with the ALWAYS_LOG line in this scope? If I read correctly it could call a bunch of observers but it looks like the log channel isn't on by default.

Reply:
By concerned I mean should I release the lock before calling ALWAYS_LOG and then re-take it again before interacting with the map

Reply:
I think this can be rewritten like so:
```
bool containedAction;
{ 
    Locker lock { m_actionHandlersLock };
    containedAction  = m_actionHandlers.remove(action)
}
if (containedAction)
    ALWAYS_LOG(LOGIDENTIFIER, "removing ", action);
```

Though it would probably be acceptable to include the ALWAYS_LOCK within the lock scope too.